File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/viewmodel/item/ItemIssueViewModel.java
package io.ganguo.demo.viewmodel.item;

import android.view.View;

import io.ganguo.demo.R;
import io.ganguo.demo.entity.IssueEntity;
import io.ganguo.vmodel.BaseViewModel;

/**
 * Created by hulkyao on 13/6/2017.
 */

public class ItemIssueViewModel extends BaseViewModel {

    private String title = null;

    public ItemIssueViewModel(IssueEntity issue) {
        this.title = issue.getTitle();
    }

    @Override
    public void onViewAttached(View view) {

    }

    public String getIssueTitle() {
        return title;
    }

    @Override
    public int getItemLayoutId() {
        return R.layout.item_issue;
    }

}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/viewmodel/item/ItemIssueViewModel.kt
package io.ganguo.demo.viewmodel.item

import android.view.View

import io.ganguo.demo.R
import io.ganguo.demo.databinding.ItemIssueBinding
import io.ganguo.demo.entity.IssueEntity
import io.ganguo.viewmodel.core.viewinterface.ViewInterface
import io.ganguo.viewmodel.core.BaseViewModel

/**
 * Created by hulkyao on 13/6/2017.
 */

class ItemIssueViewModel(issue: IssueEntity) : BaseViewModel<ViewInterface<ItemIssueBinding>>() {
    override val layoutId: Int by lazy {
        R.layout.item_issue
    }
    var issueTitle: String? = null

    init {
        this.issueTitle = issue.title
    }

    override fun onViewAttached(view: View) {

    }


}
